datafusion-federation/src/analyzer.rs

```diff
@@ -124,24 +144,25 @@ impl FederationAnalyzerRule {
 
         Ok((Some(new_plan), None))
     }
+}
 
-    fn get_federation_provider(&self, plan: &LogicalPlan) -> Result<Option<FederationProviderRef>> {
-        match plan {
-            LogicalPlan::TableScan(TableScan { ref source, .. }) => {
-                let Some(federated_source) = get_table_source(source)? else {
-                    return Ok(None);
-                };
-                let provider = federated_source.federation_provider();
-                Ok(Some(provider))
-            }
-            _ => Ok(None),
+fn get_federation_provider(plan: &LogicalPlan) -> Result<Option<FederationProviderRef>> {
+    match plan {
+        LogicalPlan::TableScan(TableScan { ref source, .. }) => {
+            let Some(federated_source) = get_table_source(source)? else {
+                return Ok(None);
+            };
+            let provider = federated_source.federation_provider();
+            Ok(Some(provider))
         }
+        _ => Ok(None),
     }
 }
 
 fn wrap_projection(plan: LogicalPlan) -> Result<LogicalPlan> {
     // TODO: minimize requested columns
     match plan {
+        //LogicalPlan::Projection(_) | LogicalPlan::TableScan(_)  => Ok(plan),
```

Comment:
?

Reply:
This will be removed in final PR, this is just temp for experimenting